### PR TITLE
#208 Autofocus Option for ais-search-box - Resolved

### LINKED ISSUE #208: Autofocus Option for ais-search-box
Could an autofocus option be added for ais-search-box?

### PR DESCRIPTION
**Pull request by kedar9444**
#208 Autofocus Option for ais-search-box - Resolved

### CODE REVIEW

File: examples/dev-novel/main.ts [hunk 1]
```diff
@@ -496,6 +496,7 @@ storiesOf("SearchBox")
     wrapWithHits({
       template: `
         <ais-search-box
+          [setFocus]="true"
```
Comment (Haroenv) [inline examples/dev-novel/main.ts]: is this normal Angular convention? I'd expect this prop to be called `focused` or `autofocus` (like the DOM prop)

Comment (tkrugg) [reply examples/dev-novel/main.ts]: If I read this right, [instantsearch.js is using autofocus](https://github.com/algolia/instantsearch.js/blob/9632ade6434fb9257bdb23c66dde8aaa1c31c4eb/src/widgets/search-box/search-box.js#L17). Maybe let use autofocus for the sake of consistency?

Comment (kedar9444) [reply examples/dev-novel/main.ts]: sure will change it accordingly.

**Comment by Haroenv** [comment]
To be a complete PR this also needs tests and documentation. However it seems like the tests of `search-box` were broken somehow, so you can leave that for afterwards if needed :)

File: examples/dev-novel/main.ts [hunk 2]
```diff
@@ -496,6 +496,7 @@ storiesOf("SearchBox")
     wrapWithHits({
       template: `
         <ais-search-box
+          [autoFocus]="true"
```
[the remark this answers is not in the document]
Comment (Haroenv) [reply examples/dev-novel/main.ts]: agreed, although link is https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/autofocus

[the remark this answers is not in the document]
Comment (kedar9444) [reply examples/dev-novel/main.ts]: 👍

File: src/search-box/search-box.ts [hunk 3]
```diff
@@ -70,11 +73,13 @@ import { noop } from "../utils";
     </div>
   `
 })
-export class NgAisSearchBox extends BaseWidget {
+export class NgAisSearchBox extends BaseWidget implements AfterViewInit {
+  @ViewChild("searchBox") searchBox: any;
```
[the remark this answers is not in the document]
Comment (kedar9444) [reply src/search-box/search-box.ts]: 👍

**Comment by tkrugg** [comment]
@kedar9444 you PR has landed 🎉  as part of #297
Thank you very much for your contribution!

**Pull request closed without merging**